File: DarkOrbit 10.0/Net/netty/commands/MessageLocalizedWildcardCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class MessageLocalizedWildcardCommand
    {
        public static short ID = 11751;

        public List<MessageWildcardReplacementModule> mWildCardReplacements;
        public String mBaseKey;
        public ClientUITooltipTextFormatModule var_3317;

        public MessageLocalizedWildcardCommand(String pBaseKey, ClientUITooltipTextFormatModule param2,
                                           List<MessageWildcardReplacementModule> pWildCardReplacements)
        {
            this.mBaseKey = pBaseKey;
            this.var_3317 = param2;
            this.mWildCardReplacements = pWildCardReplacements;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(-27064);
            param1.writeShort(-14597);
            param1.writeInt(this.mWildCardReplacements.Count);
            foreach(var rm in this.mWildCardReplacements)
            {
                param1.write(rm.write());
            }
            param1.writeUTF(this.mBaseKey);
            param1.write(var_3317.write());
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/MessageLocalizedWildcardCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;
import java.util.Vector;

import simulator.netty.ServerCommand;

/**
 Created by Pedro on 30-03-2015.
 */
public class MessageLocalizedWildcardCommand
        implements ServerCommand {

    public static int ID = 17438;

    public Vector<MessageWildcardReplacementModule> mWildCardReplacements;
    public String                                   mBaseKey;
    public ClientUITooltipTextFormatModule          var_3317;

    public MessageLocalizedWildcardCommand(String pBaseKey, ClientUITooltipTextFormatModule param2,
                                           Vector<MessageWildcardReplacementModule> pWildCardReplacements) {
        this.mBaseKey = pBaseKey;
        this.var_3317 = param2;
        this.mWildCardReplacements = pWildCardReplacements;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeInt(this.mWildCardReplacements.size());
            for (MessageWildcardReplacementModule c : this.mWildCardReplacements) {
                c.write(param1);
            }
            param1.writeUTF(this.mBaseKey);
            this.var_3317.write(param1);
            param1.writeShort(-30581);
        } catch (IOException e) {
        }
    }
}
